scripts/tools/metricExportData.yml

```diff
@@ -0,0 +1,159 @@
+details:
+    - "This structure describes the export data format."
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metrics calculation descriptor"
+version: "1.6"
+class: $tMetric
+name: $t_metric_calculate_exp_desc_t
+base: $t_base_desc_t
+members:
+    - type: uint32_t
+      name: "rawReportSkipCount"
+      desc: "[in] number of reports to skip during calculation"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Export Metrics Data for machine independent calculation."
+version: "1.6"
+class: $tMetricGroup
+name: GetExportDataExp
+decl: static
+details:
+    - "This function exports or stores raw data and necessary information to perform machine independent calculation."
+    - "Optionally, the function returns the format of exported data as an array of $t_metric_export_data_format_exp_t."
+    - "The application may call this function from simultaneous threads."
+params:
+    - type: "$t_metric_group_handle_t"
+      name: hMetricGroup
+      desc: "[in] handle of the metric group"
+    - type: "const uint8_t*"
+      name: pRawData
+      desc: "[in] buffer of raw data"
+    - type: size_t
+      name: rawDataSize
+      desc: "[in] size in bytes of raw data buffer"
+    - type: "size_t*"
+      name: pExportDataSize
+      desc: |
+            [in,out] size in bytes of export data buffer
+            if size is zero, then the driver shall update the value with the number of bytes necessary to store the exported data.
+            if size is greater than required, then the driver shall update the value with the actual number of bytes necessary to store the exported data.
+    - type: "uint32_t*"
+      name: pExportDataFormatCount
+      desc: |
+            [in,out] count of $t_metric_export_data_format_exp_t elements which describes the format of exported data.
+            if count is zero, then the driver shall update the value with the count of $t_metric_export_data_format_exp_t elements necessary to describe the exported data.
+            if count is greater than required, then the driver shall update the value with the actual number of count necessary to describe the exported data.
+    - type: "uint8_t *"
+      name: pExportData
+      desc: |
+            [in,out][optional][range(0, *pExportDataSize)] buffer of exported data.
+    - type: "$t_metric_export_data_format_exp_t *"
+      name: pExportFormatData
+      desc: |
+            [in,out][optional][range(0, *pExportDataFormatCount)] buffer of exported data format
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Calculate one or more sets of metric values from exported raw data."
+version: "1.6"
+class: $xDriver
+name: CalculateMetricExportDataExp
+decl: static
+details:
+    - "This function is similar to $tMetricGroupCalculateMultipleMetricValuesExp except it would calculate from exported metric data."
```

Comment:
IMHO would start by describing what the API does, then add the relations. That seems clearer. 

`Calculate metrics values with raw data returned by $GetExportDataExp, which can be done in different node that where data was collected. This function is similar to $tMetricGroupCalculateMultipleMetricValuesExp except it calculates from exported metric data.`

Reply:
Thanks. Please check the updated one.